code/controllers/subsystems/job.dm

```diff
@@ -403,6 +403,17 @@
 
 	INVOKE_ASYNC(GLOBAL_PROC, .proc/show_location_blurb, H.client, 30)
 
+	if(joined_late)
+		var/antag_count = 0
+		for(var/antag_type in SSticker.mode.antag_tags)
+			var/datum/antagonist/A = all_antag_types[antag_type]
+			antag_count += A.get_active_antag_count()
+		for(var/antag_type in SSticker.mode.antag_tags)
+			var/datum/antagonist/A = all_antag_types[antag_type]
+			if((A.role_type in H.client.prefs.be_special_role) && !(A.flags & ANTAG_OVERRIDE_JOB) && antag_count < A.hard_cap_round && antag_count <= (length(player_list) / SSticker.mode.antag_scaling_coeff))
+				A.add_antagonist(H.mind)
+				break
+
```

Comment:
Dont roll your own logic here to determine if additional antags can be spawned in.
Use the established procs/vars for that.
`update_current_antag_max()`
`cur_max`